tests/runner.py

```diff
@@ -490,9 +490,19 @@ def fix_target(ll_filename):
 
     return output_obj
 
-  def get_emcc_args(self):
-    # TODO(sbc): We should probably unify Building.COMPILER_TEST_OPTS and self.emcc_args
-    return self.serialize_settings() + self.emcc_args + Building.COMPILER_TEST_OPTS
+  # returns the full list of arguments to pass to emcc
+  # param @main_file whether this is the main file of the test. some arguments
+  #                  (like --pre-js) do not need to be passed when building
+  #                  libraries, for example
+  def get_emcc_args(self, main_file=False):
+    args = self.serialize_settings() + self.emcc_args
+    if not main_file:
+      for i, arg in enumerate(args):
+        if arg in ('--pre-js', '--post-js'):
+          args[i] = None
+          args[i + 1] = None
+      args = [arg for arg in args if arg is not None]
```

Comment:
Is there any harm in passing all --pre-js and --post-js?

Reply:
Should be "main_file=True" since most of our tests are single file tests that compile and link at once?

Reply:
can you rename "main_file" to something like "linking" or "include_ldflags"?

Reply:
The concrete harm I saw is that building a library may be in a totally different directory, so it can't find the pre-js file, and we hit an error. Also configure tests that run code could fail due to pre-js stuff that assumes it's in the main program.

I feel like `main_file` is clearer than `linking` or `include_ldflags`. Maybe `library` as an alternative to `linking`? But I feel like the concept here is that we do have a single main file for each test in core, so the term is natural?

Reply:
But it could be that you compile the "main.c" -> "main.o", then link "main.o + other.o" into "out.js".  To me its "am I linking" not "am I compiling the main source".      As it happens we often compile and link the main source in a single command, but it could be the other way around.

Imagine this strange: 

```
emcc -c main.c -o main.o
emcc other.c main.o -o test.js 
```

I'm not saying its likely we will see this case,  I'm just using to illustrate the fact that '--pre-js' '--post-js' are fundamentally link flags.

Reply:
There are other "executables" though, like when building test programs during a configure of a library. Maybe what we needs is a way to pass specific emcc flags for the `do_run` call itself, then it's less ambiguous. That seems like a larger refactoring though.